pkg/localcontroller/manager/dataset.go

```diff
@@ -206,6 +233,12 @@ func (dm *DatasetManager) readByLine(url string) (*DataSource, error) {
 		NumberOfSamples: numberOfSamples,
 	}
 
+	if !ds.IsLocalStorage {
```

Comment:
here again having if-judge about local, better to place into storage.go

Reply:
fix it